hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -324,6 +331,24 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         "Committed to table {} with the new metadata location {}", fullName, newMetadataLocation);
   }
 
+  private BaseMetastoreOperations.CommitStatus handlePossibleConcurrentModification(
+      String newMetadataLocation, TableMetadata metadata) {
+    Optional<Boolean> locationCommitted =
+        metadataLocationCommitted(
+            tableName(),
+            newMetadataLocation,
+            metadata.properties(),
+            () -> checkCurrentMetadataLocation(newMetadataLocation));
+    if (locationCommitted.isPresent()) {
+      if (locationCommitted.get()) {
+        return BaseMetastoreOperations.CommitStatus.SUCCESS;
+      } else {
+        return BaseMetastoreOperations.CommitStatus.FAILURE;
+      }
+    }
+    return BaseMetastoreOperations.CommitStatus.UNKNOWN;
+  }
+
```

Comment:
We can get rid of this method, if we have something like `checkCommitStatusStrict` method